controllers/remote/cluster.go

```diff
@@ -29,7 +29,7 @@ import (
 )
 
 const (
-	defaultClientTimeout = 10 * time.Second
+	defaultClientTimeout = 0 * time.Second
```

Comment:
I'm not sure if this is the right approach, but it is the setting on the reconciler client which talks to the management cluster. 

I think we should be happy to at least increase this default e.g. to 120 seconds and see what impact this could have when testing at scale with faults.

This timeout was added in https://github.com/kubernetes-sigs/cluster-api/issues/2993 @fabriziopandini do you know in what circumstances this can hang, and why it doesn't seem to happen with the management cluster.

Reply:
>  to the management cluster.

mgmt or workload cluster?

Reply:
What is the effective default we use for the mgmt cluster? (CR or client-go I guess)

Reply:
Just remembered that we have to be very careful with this timeout. Basically if a workload cluster is down with our current implementation an entire worker is stuck in trying to create a client.

Might be better implementation options though

Reply:
I share the same concern.

Also, AFAIK having a timeout doesn't seem an issue with a real API server, so it might be something related to the watch implementation in the in-memory provider

Reply:
IIRC what Killian mentioned it is a problem with the real API server. All our watches to workload clusters are getting closed after ~ 10-15 seconds

Reply:
This PR is not related to the in-memory provider. Today when CAPI opens watches against a workload API Server those watches are closed after 10 seconds, meaning many more watches are opened and closed than is strictly necessary. 

I'm not sure if there's a major performance penalty from opening that many watches - but it's multiple watches per workload cluster so I'd imagine it could be significant.

Reply:
What about changing the cluster cache tracker so we have a dedicated client for watches with a bigger timeout, like e.g 5 or 10 minutes?
The alternative is to use something more than 10s, that suites better the watch use case, but not too high because this is also the duration we block all reconciles in case of connectivity issues.... definitely not an easy trade-off